qctrlopencontrols/driven_controls/predefined.py

```diff
@@ -812,3 +813,129 @@ def _new_walsh_amplitude_modulated_filter_1_control(
         durations=durations,
         **kwargs,
     )
+
+
+def new_modulated_gaussian_control(
+    maximum_rabi_rate: float,
+    minimum_segment_duration: float,
+    duration: float,
+    modulation_frequency: float,
+) -> DrivenControl:
+    """
+    Generate a Gaussian driven control sequence modulated by a sinusoidal signal at a specific
+    frequency.
+
+    The net effect of this control sequence is an identity gate.
+
+    Parameters
+    ----------
+    maximum_rabi_rate: float
+        Maximum Rabi rate of the system.
+
+    minimum_segment_duration : float
+        Minimum length of each segment in the control sequence.
+
+    duration : float
+        Total duration of the control sequence.
+
+    modulation_frequency: float
+        Frequency of the modulation sinusoidal signal.
+
+    Returns
+    -------
+    DrivenControl
+        A control sequence as an instance of DrivenControl.
+    """
+
+    check_arguments(
+        maximum_rabi_rate > 0.0,
+        "Maximum Rabi rate must be greater than zero.",
+        {"maximum_rabi_rate": maximum_rabi_rate},
+    )
+
+    check_arguments(
+        minimum_segment_duration > 0.0,
+        "Minimum segment duration must be greater than zero.",
+        {"minimum_segment_duration": minimum_segment_duration},
+    )
+
+    check_arguments(
+        duration > minimum_segment_duration,
+        "Total duration must be greater than minimum segment duration.",
+        {"duration": duration, "minimum_segment_duration": minimum_segment_duration,},
+    )
+
+    # default spread of the gaussian shaped pulse as a fraction of its duration
+    _pulse_width = 0.1
+
+    # default mean of the gaussian shaped pulse as a fraction of its duration
+    _pulse_mean = 0.5
+
+    min_required_upper_bound = np.sqrt(2 * np.pi) / (_pulse_width * duration)
+    check_arguments(
+        maximum_rabi_rate >= min_required_upper_bound,
+        "Maximum Rabi rate must be large enough to permit a 2Pi rotation.",
+        {"maximum_rabi_rate": maximum_rabi_rate},
+        extras={
+            "minimum required value for upper_bound "
+            "(sqrt(2pi)/(0.1*maximum_duration))": min_required_upper_bound
+        },
+    )
+
+    segment_start_times = np.arange(0, duration, minimum_segment_duration)
```

Comment:
Given that it's a specified `duration` but only a *minimum* segment duration, I think instead we should use slightly longer segment lengths so that we get a pulse of duration `duration` exactly:
```
segment_count = int(np.ceil(duration / minimum_segment_duration))
segment_start_times = np.arange(segment_count) * duration / segment_count # or similar
...
```

Reply:
That's right, sorry I missed that point. Fixed.